local-cluster/tests/local_cluster.rs

```diff
@@ -1341,6 +1341,8 @@ fn test_optimistic_confirmation_violation() {
     // Mark fork as dead on the heavier validator, this should make the fork effectively
     // dead, even though it was optimistically confirmed. The smaller validator should
     // jump over to the new fork
+    // Also, remove saved tower to intentionally make the restarted validator to violate the
+    // optimistic confirmation
```

Comment:
I'm not fully sure but it seems that this test (added by #11353) doesn't get along with persistent tower... Maybe, because of testing some failure case?

Reply:
@ryoqun ah yeah! That's actually good this test didn't work, it means the persistent tower is serving it's purpose 😄 

What's essentially happening in this test is we are trying to simulate a vote that would an optimistic confirmation violation. So we have two validators A with 51% of the stake, B with 49%:

```
1-----2 (Vote A, B, optimistically confirmed) ---3
```

 We then kill validator `A`, mark slot `2` as dead and restart. Without persistent tower, this means validator `A` will make its own fork like so:

```
1-----2 (Vote A, B, optimistically confirmed) ---3
  |
   -----------------------4 (Validator A)
```

And validator `A` will then vote on slot 4 and b/c it has 51% of the stake, validator `B` will switch over as well and roots will be made on this new fork. This is bad because the `optimistically confirmed` slot `2` was not rooted (rolled back)

However, with this PR, `A` will not be able to vote on slot `4` because it remembers it's last vote was on slot `2`, and because `B` is still voting on `fork 2`, `A` will not be able to generate a switching proof to slot `4`. Hence this test will stall!

Reply:
It might actually be cool to add this as another version of this test! If we don't remove the tower, we should assert that the cluster makes no progress with `cluster.check_no_new_roots`

Reply:
Ok, I'll work on this!